sdk/src/client/api/block_builder/input_selection/remainder.rs

```diff
@@ -118,30 +122,77 @@ impl InputSelection {
 
         let native_tokens_diff = get_native_tokens_diff(&input_native_tokens, &output_native_tokens)?;
 
-        if inputs_sum == outputs_sum && native_tokens_diff.is_none() {
+        let mut input_mana = 0;
+
+        for input in &self.selected_inputs {
+            input_mana += input.output.all_mana(
+                &self.protocol_parameters,
+                input.output_id().transaction_id().slot_index(),
+                self.slot_index,
+            )?;
+            // TODO rewards https://github.com/iotaledger/iota-sdk/issues/1310
+        }
+
+        let output_mana = self.outputs.iter().map(|o| o.mana()).sum::<u64>() + self.mana_allotments;
+
+        if input_amount == output_amount && input_mana == output_mana && native_tokens_diff.is_none() {
             log::debug!("No remainder required");
             return Ok((None, storage_deposit_returns));
         }
 
+        let amount_diff = input_amount - output_amount;
+        let mana_diff = input_mana - output_mana;
+
+        // If there is only a mana remainder, try to fit it in an automatically transitioned output.
+        if input_amount == output_amount && input_mana != output_mana && native_tokens_diff.is_none() {
+            let output = self
+                .outputs
+                .iter_mut()
+                .filter(|output| {
+                    output
+                        .chain_id()
+                        .as_ref()
+                        .map(|chain_id| self.automatically_transitioned.contains(chain_id))
+                        .unwrap_or(false)
+                        // Foundries can't hold mana so they are not considered here.
+                        && !output.is_foundry()
+                })
+                .next();
```

Comment:
I guess we should prefer account outputs here instead of just using the first one

Reply:
I'll do something but it's not pretty because we can't clone mutable iterators

Reply:
Maybe something with partition by type would work?

Reply:
I was trying to find a way that doesn't require collecting

Reply:
We can also open an issue and merge like this for now, if you want